### ISSUE TITLE
How to preduce the trimap?

### ISSUE THREAD

**Issue by xaibeing** [foamliu/Deep-Image-Matting#26]
I do some test but didn't get result as good as the images shown in the README.md.
I think trimap have a huge impact to the result. I use Mask RCNN to generator segmentation and than the trimap.
Could you tell how do you produce the trimap in the README.md  [demo images](https://github.com/foamliu/Deep-Image-Matting/blob/master/README.md#demo) ?
Thanks!

**Comment by tsing90** [comment]
good question, refer to #5

**Comment by rainsun1** [comment]
You can use the "Robust Matting" demo to produce the trimap.

**Comment by tsing90** [comment]
@rainsun1 could you put a link of code here so that we can have a try? I searched for the code, it seemed 'robust matting' algorithm also need trimap to produce final result.

**Comment by rainsun1** [comment]
Here is a robust matting demo, you could load an image and then create a trimap yourself by the brushes and inpainting.

[RobustMatting_1.45.zip](https://github.com/foamliu/Deep-Image-Matting/files/2844890/RobustMatting_1.45.zip)

**Comment by tsing90** [comment]
@rainsun1 many thanks for sharing this, actually, it'll be better if the trimap can be automatically generated so that the whole process can be automatic. recently, i found a useful website which can do bg removal all in automation: http://remove.bg

**Comment by rainsun1** [comment]
@tsing90 A paper named "Semantic Human Matting" is in this automatic trimap way.

**Comment by tsing90** [comment]
@rainsun1 many thanks, that's what I need. unfortunately, the big datasets shown in that paper are not available for us. that kind of big datasets is essential for training a good model.

**Comment by rainsun1** [comment]
@tsing90 yes, but can you find some other implementations on "Semantic Human Matting" on github?

**Comment by wodsoe** [comment]
remove.bg,so wonderful

**Comment by tsing90** [comment]
@rainsun1 yes, I found some, like 
https://github.com/lizhengwei1992/Semantic_Human_Matting
https://github.com/CapsulE07/semantic-human-matting

**Comment by tsing90** [comment]
@wodsoe that's our benchmark from industry ;)

**Comment by wodsoe** [comment]
> @wodsoe that's our benchmark from industry ;)
How to re-implement

**Comment by farazBhatti** [comment]
@tsing90 you can use Deep Lab v3+(https://github.com/tensorflow/models/blob/master/research/deeplab/deeplab_demo.ipynb) to get segmentation mask and then apply trimap generator to produce trimaps.
https://github.com/lnugraha/trimap_generator

**Comment by tsing90** [comment]
@farazBhatti thanks for your suggestion. actually I've already tried its xception model which is good to get decent masks. To get trimaps, it's enough just by doing dilating and eroding. The key problem is that the masks we got from semantic segmentation (like deeplab v3+) is binary, its quality is lower than alpha matting

**Comment by farazBhatti** [comment]
@tsing90, yes true we only need medium quality segmentation mask to get trimap and from that trimap along with original colored image alpha matte is predicted. Alpha matte segmented image will always be superior to any other segmented mask , but we really don't need that good quality of segmented mask to predict alpha matte.

**Comment by tsing90** [comment]
Does anyone have this problem? when producing trimap using the alpoha matte image from Adobe's DIM datasets, the trimap is kind of jagged, shown like below. My first guess of this issue is due to the image quality of .jpg, because .png images in that datasets don't have this problem.
![1-1252426161dfxy](https://user-images.githubusercontent.com/20683656/53577023-d22a3900-3b6c-11e9-9be7-a3e2ed191ee1.jpg)![woman_a](https://user-images.githubusercontent.com/20683656/53577037-d8201a00-3b6c-11e9-80f7-0fd54ecdedf6.png)

**Comment by farazBhatti** [comment]
Yes it could be the result of jpg compression quality. By the way is there any link to this data(
Adobe's DIM dataset) set which you are using?Thanks

**Comment by tsing90** [comment]
@farazBhatti I got them from the author, due to license issue, you have to contact the author of that paper

**Comment by foamliu** [comment]
@xaibeing The trimaps are in the dataset obtained from the author.

**Comment by FantasyJXF** [comment]
@tsing90 Hi, I'm re-implementing the work nowadays. Could you please share some details of your tests. Hope to have a talk, my email  931026752@qq.com

**Comment by yxt132** [comment]
> @xaibeing The trimaps are in the dataset obtained from the author.

Only the test dataset has trimaps. The training set does not have them. @foamliu Can you explain a little bit how you generated trimaps for your training set? Thank you.

**Comment by sergiomena** [comment]
@xaibeing Hi, I'm also doing segmentation with mask RCNN and now I'm trying to use image matting to improve edge identification. I would like to private message you because now I have some trouble and maybe discussing could help both, if you want this is my email: sergio.17mena@gmail.com
